advanced-cache.php

```diff
@@ -425,10 +441,7 @@ function add_debug_html_to_output( $debug_html ) {
 
 // Things that define a unique page.
 if ( isset( $_SERVER['QUERY_STRING'] ) ) {
-	parse_str($_SERVER['QUERY_STRING'], $batcache->query);
-
-	// Normalize query paramaters for better cache hits.
-	ksort( $batcache->query );
+	$batcache->set_query( $_SERVER['QUERY_STRING'] );
```

Comment:
Since what we're really concerned with is how the cache key is derived from the query string, it would be more explicit (and thus perhaps better) to continue setting `$query` the previous way and change how we set the "query" part of the key [here](https://github.com/Automattic/batcache/pull/92/files#diff-75c8dd5448198d72ea18d5a910000e2265c89060674a09498f681c63b3f39ab8R451) instead.

Reply:
I wanted to move the custom logic inside the class. It's part of the internal logic of the class. Tbh, the fact that we have all these scattered pieces of code outside the class isn't making me too happy, but step-by-step. :)

Also, you didn't link correctly, so no idea what line is this pointing to:

> and change how we set the "query" part of the key here instead.

Reply:
Also, makes this thing testable. :) Or at least this new thing. I have plans for tests, tho not yet...

Reply:
That's strange. The link sort of works if you expand some unrevealed code in the diff, but it does not work by default. Here is the line I meant to reference:
https://github.com/Automattic/batcache/blob/3209ad7249a79d5f7ef3d7fbf03eceb74248669f/advanced-cache.php#L438

The idea is to explicitly change key generation itself rather than indirectly change key generation by changing its inputs. I think doing so is a stronger approach and wanted to share that thought.

I think I agree about the loose pieces outside the class, but there's nothing preventing key generation from being moved to a class method.